staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -24,7 +24,8 @@ type CustomResourceDefinitionSpec struct {
 	Group string `json:"group" protobuf:"bytes,1,opt,name=group"`
 	// Version is the version this resource belongs in
 	// Should be always first item in Versions field if provided.
-	Version string `json:"version" protobuf:"bytes,2,opt,name=version"`
+	// If Versions field is provided, this field is optional.
+	Version string `json:"version,omitempty" protobuf:"bytes,2,opt,name=version"`
```

Comment:
don't like this being optional. Would prefer to have an enforcement of `Versions[0]` and `Version` to match.

Reply:
The plan is to remove this field in the next version of the API, it make sense to make it optional I guess but I can go both ways.